### PR TITLE
fix(docs-infra): render `deprecated` markers for CLI command options

### LINKED ISSUE #27423: AIO: show note near deprecated CLI options
# 📚 Docs or angular.io bug report

### Description

At the moment when CLI command options are deprecated no indication is shown in AIO.

For instance in https://next.angular.io/cli/build certain options are deprecated. One of them being: `evalSourceMap`. 

The data is already provided in the JSON help https://github.com/angular/cli-builds/blob/master/help/build.json#L138

## Expectation
We should indicate that this is deprecated, and if there is a depreciation message instead of a `boolean`, we should show the message maybe.

### LINKED ISSUE #27563: fix(docs-infra): render `deprecated` markers for CLI command options
_Originally posted by @petebacondarwin in https://github.com/angular/angular/pull/27556#issuecomment-445466437_

### PR DESCRIPTION
**Pull request by wKoza**
fixes #27563
fixes #27423

## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

Show a deprecated label near option cli deprecated 

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: #27563 #27423

No deprecated information near cli options that were deprecated. The information exists however in build.json file.

```json
{
      "name": "vendorSourceMap",
      "description": "Resolve vendor packages sourcemaps.",
      "type": "boolean",
      "default": false,
      "required": false,
      "aliases": [],
      "hidden": false,
      "deprecated": true
    },
```

## What is the new behavior?

We have a deprecated label and we can add a message below whether we change `true`by a text:

```json
{
      "name": "vendorSourceMap",
      "description": "Resolve vendor packages sourcemaps.",
      "type": "boolean",
      "default": false,
      "required": false,
      "aliases": [],
      "hidden": false,
      "deprecated": "lorem ipsum ..."
    },
```

## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

File: aio/tools/transforms/templates/cli/lib/cli.html [hunk 1]
```diff
@@ -56,6 +56,7 @@
         <code class="cli-option-syntax no-auto-link">{$ renderOption(option.name, option.type, option.default, option.enum) $}</code>
       </td>
       <td>
+        {% if option.deprecated !== undefined and option.deprecated !== false %}<label class="raised cli-deprecated">deprecated</label> {% if option.deprecated !== true %} {$ option.deprecated | marked $} {% endif %}{% endif %}
```
Comment (petebacondarwin) [inline aio/tools/transforms/templates/cli/lib/cli.html]: In the CLI code base the `deprecated` value is:

```ts
  /**
   * Deprecation. If this flag is not false a warning will be shown on the console. Either `true`
   * or a string to show the user as a notice.
   */
  deprecated?: boolean | string;
```

So I think it is actually enough to check `{% if option.deprecated %}`, right? If it is `undefined` or `false` then this `if` will fail. Or are we concerned about the empty string: `""` - @hansl ?

**Review by petebacondarwin — CHANGES_REQUESTED** [review]
I think that the styling you have chosen for deprecated CLI options is a little to strong:

screenshot 2019-01-13 at 19 17 24

While we do use the red deprecated badge in the top heading for a whole page, we only use ~~crossed through~~ styling on the element name when referring to a deprecated within a page, with a bold **Deprecated:** marker followed by the deprecated message.

screenshot 2019-01-13 at 19 16 38

I think we should keep this in the same style.

**Comment by wKoza** [comment]
About the style, I've been inspired by the [read-only label](https://angular.io/api/forms/AbstractControl#properties) . But, I agree that it could be shrill. I'll change this.

**Comment by wKoza** [comment]
So, with the actual system of markdown renderer, we cannot have 'Deprecated' and a string on the same line. Have 'Deprecated' and 'its explain' on two separate lines  is ambiguous.

Either we have a boolean 

```json
deprecated: true
```
and it'll display **Deprecated** 

or we have a string and we have to set 

```json
deprecated: "**Deprecated:** lorem ipsum"
```
and it'll display **Deprecated:** lorem ipsum 

It's ok for you ?

**Comment by petebacondarwin** [comment]
Check out this template: https://github.com/angular/angular/blob/18b6d580c54db208ebb8b1bb4168b53393257b01/aio/tools/transforms/templates/api/lib/memberHelpers.html#L51-L54

File: aio/tools/transforms/templates/cli/lib/cli.html [hunk 2]
```diff
@@ -56,7 +56,14 @@
...
       </td>
       <td>
-        {% if option.deprecated !== undefined and option.deprecated !== false %}<label class="raised cli-deprecated">deprecated</label> {% if option.deprecated !== true %} {$ option.deprecated | marked $} {% endif %}{% endif %}
+        {% if option.deprecated %}
+          {% if option.deprecated !== true %}
+            {$ ('**Deprecated:** ' + option.deprecated) | marked $}
+          {% endif %}
+          {% if option.deprecated === true %}
+            {$ ('**Deprecated**') | marked $}
+          {% endif %}
```
Comment (petebacondarwin) [inline aio/tools/transforms/templates/cli/lib/cli.html]: Even simpler:

```
{% if option.deprecated === true %}
  <strong>Deprecated</strong>
{% else %}
  {$ ('**Deprecated:** ' + option.deprecated) | marked $}
{% endif %}
```

**Review by petebacondarwin — CHANGES_REQUESTED** [review]
Getting there! I still think that we should cross through the syntax text of the deprecated item:

screenshot 2019-01-14 at 14 52 33

File: aio/tools/transforms/templates/cli/lib/cli.html (hunk 2 above)
Comment (wKoza) [reply aio/tools/transforms/templates/cli/lib/cli.html]: I was not aware of `{% else %}` ; It's better indeed ;)

**Review by petebacondarwin — APPROVED** [review]
Awesome. Thanks @wKoza again!

File: aio/src/styles/2-modules/_cli-pages.scss [hunk 3]
```diff
@@ -7,4 +7,3 @@
 .cli-option-syntax {
   white-space: pre;
 }
-
```
Comment (gkalpak) [inline aio/src/styles/2-modules/_cli-pages.scss]: 😭

**Pull request closed without merging**